lib/media/drm_engine.js

```diff
@@ -695,14 +695,18 @@ shaka.media.DrmEngine = class {
     // Create sessions.
     const initDatas =
         (this.currentDrmInfo_ ? this.currentDrmInfo_.initData : []) || [];
-    for (const initDataOverride of initDatas) {
+    // All initDatas with initData with a size of 0 will get ignored by the
```

Comment:
I'm not certain doing the same filter in two places makes sense.  What happens when we have another reason to skip an init data?  It would be nice if newInitData() were responsible for all of it.

I think you can resolve this hang by adding a call to this.areAllSessionsLoaded_() instead of checking initDatas.length.

Reply:
Yeah, that makes sense.
I also added a test.